src/rpc/net.cpp

```diff
@@ -106,82 +106,83 @@ static RPCHelpMan ping()
+                    {RPCResult::Type::STR_HEX, "services", "The services offered"},
+                    {RPCResult::Type::ARR, "servicesnames", "the services offered, in human-readable form",
+                    {
+                        {RPCResult::Type::STR, "SERVICE_NAME", "the service name if it is recognised"}
                     }},
-                },
-                RPCExamples{
-                    HelpExampleCli("getpeerinfo", "")
+                    {RPCResult::Type::BOOL, "relaytxes", "Whether peer has asked us to relay transactions to it"},
+                    {RPCResult::Type::NUM_TIME, "lastsend", "The " + UNIX_EPOCH_TIME + " of the last send"},
+                    {RPCResult::Type::NUM_TIME, "lastrecv", "The " + UNIX_EPOCH_TIME + " of the last receive"},
+                    {RPCResult::Type::NUM_TIME, "last_transaction", "The " + UNIX_EPOCH_TIME + " of the last valid transaction received from this peer"},
+                    {RPCResult::Type::NUM_TIME, "last_block", "The " + UNIX_EPOCH_TIME + " of the last block received from this peer"},
+                    {RPCResult::Type::NUM, "bytessent", "The total bytes sent"},
+                    {RPCResult::Type::NUM, "bytesrecv", "The total bytes received"},
+                    {RPCResult::Type::NUM_TIME, "conntime", "The " + UNIX_EPOCH_TIME + " of the connection"},
+                    {RPCResult::Type::NUM, "timeoffset", "The time offset in seconds"},
+                    {RPCResult::Type::NUM, "pingtime", /* optional */ true, "ping time (if available)"},
+                    {RPCResult::Type::NUM, "minping", /* optional */ true, "minimum observed ping time (if any at all)"},
+                    {RPCResult::Type::NUM, "pingwait", /* optional */ true, "ping wait (if non-zero)"},
+                    {RPCResult::Type::NUM, "version", "The peer version, such as 70001"},
+                    {RPCResult::Type::STR, "subver", "The string version"},
+                    {RPCResult::Type::BOOL, "inbound", "Inbound (true) or Outbound (false)"},
+                    {RPCResult::Type::BOOL, "bip152_hb_to", "Whether we selected peer as (compact blocks) high-bandwidth peer"},
+                    {RPCResult::Type::BOOL, "bip152_hb_from", "Whether peer selected us as (compact blocks) high-bandwidth peer"},
+                    {RPCResult::Type::NUM, "startingheight", /*optional=*/true, "The starting height (block) of the peer"},
+                    {RPCResult::Type::NUM, "synced_headers", /*optional=*/true, "The last header we have in common with this peer"},
+                    {RPCResult::Type::NUM, "synced_blocks", /*optional=*/true, "The last block we have in common with this peer"},
+                    {RPCResult::Type::ARR, "inflight", /*optional=*/true, "",
+                    {
+                        {RPCResult::Type::NUM, "n", "The heights of blocks we're currently asking from this peer"},
+                    }},
```

Comment:
When are these not provided? Looks impossible to actually happen? Maybe a rare threading race (should we even document those?)?

Reply:
I am happy to review a pull request that always provides them.

Reply:
My point is that the current code appears to already always provide them.

Reply:
They are not if `fStateStats` is false.

Again, if you find a way to remove the `fStateStats` bool, please ping me for review.

Reply:
When can it be false?

Reply:
> When can it be false?

You answered this in your first comment:

> threading race (should we even document those?)?

I'd say yes, not only to clarify the interface, but also to avoid applications from crashing due to unexpected KeyError